src/tracks/MidiClip.cpp

```diff
@@ -208,23 +208,38 @@ Note * MidiClip::addNote( const Note & _new_note, const bool _quant_pos )
 
 
 
-void MidiClip::removeNote( Note * _note_to_del )
+NoteVector::const_iterator MidiClip::removeNote(NoteVector::const_iterator note)
 {
 	instrumentTrack()->lock();
+	delete *note;
+	auto new_it = m_notes.erase(note);
+	instrumentTrack()->unlock();
+
+	checkType();
+	updateLength();
 
-	m_notes.erase(std::remove_if(m_notes.begin(), m_notes.end(), [&](Note* note)
+	emit dataChanged();
+	return new_it;
+}
+
+NoteVector::const_iterator MidiClip::removeNote(Note* note_to_del )
+{
+	instrumentTrack()->lock();
+
+	auto it = std::find(m_notes.begin(), m_notes.end(), note_to_del);
```

Comment:
```suggestion
NoteVector::const_iterator MidiClip::removeNote(Note* note)
{
	instrumentTrack()->lock();

	auto it = std::find(m_notes.begin(), m_notes.end(), note);
```
Might as well fix the style too

Reply:
The variable name is fine imo. I already deleted the _ prefix.

Reply:
We don't use snake case though.

Reply:
I completely forgot that. Good point.